pkg/v7/configmap/desired.go

```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/giantswarm/cluster-operator/pkg/v7/key"
 )
 
-func (s *Service) GetDesiredState(ctx context.Context, clusterConfig ClusterConfig, configMapValues ConfigMapValues) ([]*corev1.ConfigMap, error) {
+func (s *Service) GetDesiredState(ctx context.Context, clusterConfig ClusterConfig, configMapValues ConfigMapValues, providerChartSpecs []key.ChartSpec) ([]*corev1.ConfigMap, error) {
```

Comment:
Why is that tho?

Reply:
This is so that per provider config can be injected. For chartconfigs this is already in place.

e..g `external-dns` chart is only needed for Azure.
https://github.com/giantswarm/cluster-operator/blob/475ae0f55bce3f7c3429570eb0a04ae1592609ee/service/controller/azure/v7/key/key.go#L11-L22

The spec is passed to the chartconfig service.
https://github.com/giantswarm/cluster-operator/blob/475ae0f55bce3f7c3429570eb0a04ae1592609ee/service/controller/azure/v7/resource/chartconfig/desired.go#L31

The same support is needed for configmaps.